Sources/SwiftKafka/Utilities/Duration+Helpers.swift

```diff
@@ -0,0 +1,25 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the swift-kafka-gsoc open source project
+//
+// Copyright (c) 2023 Apple Inc. and the swift-kafka-gsoc project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-kafka-gsoc project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+extension Duration {
+    internal var inMilliseconds: UInt {
+        let seconds = Double(components.seconds) * 1000.0
+        let attoseconds = Double(components.attoseconds) * 1e-15
+        return UInt(seconds + attoseconds)
+    }
+
+    internal var canBeRepresentedAsMilliseconds: Bool {
+        return self.inMilliseconds > 0
```

Comment:
Not sure if it is right place but in some places zero is a valid milliseconds as well
should be `self == .zero || self.inMilliseconds > 0`?

Reply:
I would keep it like this for now and if we come across a concrete use-case where we need 0 then we should evaluate this again.